testool/src/main.rs

```diff
@@ -0,0 +1,222 @@
+    );
+
+    println!("Execution result is : {:?}", result);
+
+    Ok(())
+}
+
+fn main() -> Result<()> {
+    //  RAYON_NUM_THREADS=1 RUST_BACKTRACE=1 cargo run -- --path
+    // "tests/src/GeneralStateTestsFiller/**/" --skip-state-circuit
+
+    let mut config = Config::load()?;
+    let args = Args::parse();
+
+    let bytecode_test_config = BytecodeTestConfig {
+        enable_state_circuit_test: !args.skip_state_circuit,
+        ..Default::default()
+    };
+
+    let statetest_config = StateTestConfig {
+        run_circuit: !args.skip_circuit,
+        bytecode_test_config: bytecode_test_config.clone(),
+        global: config.clone(),
+    };
+
+    if let Some(raw) = &args.raw {
+        run_bytecode(raw, bytecode_test_config)?;
+        return Ok(());
+    }
+
+    env_logger::Builder::from_env(env_logger::Env::default().default_filter_or("info")).init();
+
+    log::info!("Parsing and compliling tests...");
+    let compiler = Compiler::new(true, Some(PathBuf::from("./code.cache")))?;
+
+    if let Some(test_id) = args.ethtest_id {
+        // test only one
+        config.skip_test.clear();
+        let state_tests = load_statetests_suite(&args.ethtest, config, compiler)?;
+        let mut state_tests: Vec<_> = state_tests
+            .into_iter()
+            .filter(|t| t.id == test_id)
+            .collect();
+        if state_tests.is_empty() {
+            bail!("test '{}' not found", test_id);
+        }
+        run_single_test(state_tests.remove(0), statetest_config)?;
+    } else if args.ci {
+        // ci mode
+        config.skip_test.clear();
+        let path = "tests/src/GeneralStateTestsFiller/**/*";
+
+        let state_tests = load_statetests_suite(path, config, compiler)?;
+        let output = Command::new("git")
+            .args(&["rev-parse", "HEAD"])
+            .output()
+            .unwrap();
+        let git_hash = String::from_utf8(output.stdout).unwrap();
+        let git_hash = &git_hash[..7].to_string();
+        let timestamp = SystemTime::now()
+            .duration_since(SystemTime::UNIX_EPOCH)
+            .unwrap()
+            .as_secs();
+
+        std::fs::create_dir_all(REPORT_FOLDER)?;
+        let csv_filename = format!("{}/{}.{}.csv", REPORT_FOLDER, timestamp, git_hash);
+        let html_filename = format!("{}/{}.{}.html", REPORT_FOLDER, timestamp, git_hash);
+
+        let mut results = Results::with_cache(PathBuf::from(csv_filename))?;
+        run_statetests_suite(state_tests, statetest_config, &mut results)?;
+
+        // filter non-csv files and files from the same commit
+        let mut files: Vec<_> = std::fs::read_dir(REPORT_FOLDER)
+            .unwrap()
+            .filter_map(|f| {
+                let filename = f.unwrap().file_name().to_str().unwrap().to_string();
+                if filename.ends_with(".csv") && !filename.contains(&format!(".{}.", git_hash)) {
+                    Some(filename)
+                } else {
+                    None
+                }
```

Comment:
We have something like `bool::then` or `bool::then_some` for these
```
assert_eq!(false.then_some(0), None);
assert_eq!(true.then_some(0), Some(0));
```
```
assert_eq!(false.then(|| 0), None);
assert_eq!(true.then(|| 0), Some(0));
```
```suggestion
                (filename.ends_with(".csv") && !filename.contains(&format!(".{}.", git_hash))).then_some(filename)
```